### PR TITLE
Use optimistic updates

### PR DESCRIPTION
**Pull request by semtexzv**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: evaluator/evaluate.go [hunk 1]
```diff
@@ -117,6 +117,7 @@ func evaluateWithVmaas(ctx context.Context, tx *gorm.DB, updatesReq *vmaas.Updat
 	}
 	system.ThirdParty = thirdParty                    // to set "system_platform.third_party" column
 	updatesReq.ThirdParty = utils.PtrBool(thirdParty) // enable "third_party" updates in VMaaS if needed
+	updatesReq.OptimisticUpdates = utils.PtrBool(true)
```
Comment (Josca) [inline evaluator/evaluate.go:120]: We need that for EPEL only, right? If EPEL is third-party we need to enable it only on third-party enabled.

**Review by Josca — CHANGES_REQUESTED** [review]

File: evaluator/evaluate.go (hunk 1 above)
Comment (semtexzv) [reply evaluator/evaluate.go]: Fixed

File: evaluator/evaluate.go [hunk 2]
```diff
@@ -117,8 +117,9 @@ func evaluateWithVmaas(ctx context.Context, tx *gorm.DB, updatesReq *vmaas.Updat
 	}
 	system.ThirdParty = thirdParty                    // to set "system_platform.third_party" column
 	updatesReq.ThirdParty = utils.PtrBool(thirdParty) // enable "third_party" updates in VMaaS if needed
+	updatesReq.OptimisticUpdates = utils.PtrBool(thirdParty)
 
-	vmaasData, err := callVMaas(ctx, updatesReq)
+	vmaasData, err := callVMaas(ctx, oupdatesReq)
```
Comment (Josca) [inline evaluator/evaluate.go:122]: typo

**Review by Josca — CHANGES_REQUESTED** [review]

File: base/base.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ func init() {
 }
 
 func HandleSignals() {
-	c := make(chan os.Signal)
+	c := make(chan os.Signal, 1)
```
Comment (Josca) [inline base/base.go:28]: why are we updating this?

Comment (semtexzv) [reply base/base.go:28]: Lint started failing.

**Pull request merged**